packages/test/test-end-to-end-tests/src/test/handleValidation.spec.ts

```diff
@@ -0,0 +1,425 @@
+		[mapId, SharedMap.getFactory()],
+		[stringId, SharedString.getFactory()],
+		[cellId, SharedCell.getFactory()],
+		[counterId, SharedCounter.getFactory()],
+		[directoryId, SharedDirectory.getFactory()],
+		[treeId, SharedTree.getFactory()],
+		[matrixId, SharedMatrix.getFactory()],
+		// [legacyTreeId, LegacySharedTree.getFactory()],
+		[registerId, ConsensusRegisterCollection.getFactory()],
+		[queueId, ConsensusQueue.getFactory()],
+		// [
+		// 	migrationShimId,
+		// 	new MigrationShimFactory(
+		// 		LegacySharedTree.getFactory(),
+		// 		SharedTree.getFactory(),
+		// 		(legacyTree, newTree) => {
+		// 			throw new Error("unreachable");
+		// 		},
+		// 	),
+		// ],
+	];
+
+	const testContainerConfig: ITestContainerConfig = {
+		fluidDataObjectType: DataObjectFactoryType.Test,
+		registry,
+		runtimeOptions: {
+			enableRuntimeIdCompressor: "on",
+		},
+		loaderProps: {
+			configProvider: configProvider({
+				"Fluid.Container.enableOfflineLoad": true,
+			}),
+		},
+	};
+
+	async function setup() {
+		const provider = getTestObjectProvider();
+		const loader = provider.makeTestLoader(testContainerConfig);
+		const container1 = await createAndAttachContainer(
+			provider.defaultCodeDetails,
+			loader,
+			provider.driver.createCreateNewRequest(provider.documentId),
+		);
+		provider.updateDocumentId(container1.resolvedUrl);
+		const url = await container1.getAbsoluteUrl("");
+		const dataStore1 = (await container1.getEntryPoint()) as ITestFluidObject;
+		const map1 = await dataStore1.getSharedObject<ISharedMap>(mapId);
+		const cell1 = await dataStore1.getSharedObject<ISharedCell>(cellId);
+		const directory1 = await dataStore1.getSharedObject<SharedDirectory>(directoryId);
+		const tree1 = await dataStore1.getSharedObject<ISharedTree>(treeId);
+		const matrix1 = await dataStore1.getSharedObject<ISharedMatrix>(matrixId);
+		// legacyTree1 = await dataStore1.getSharedObject<LegacySharedTree>(legacyTreeId);
+		const register1 =
+			await dataStore1.getSharedObject<IConsensusRegisterCollection>(registerId);
+		const queue1 = await dataStore1.getSharedObject<IConsensusOrderedCollection>(queueId);
+		// migrationShim1 = await dataStore1.getSharedObject<MigrationShim>(migrationShimId);
+		const string1 = await dataStore1.getSharedObject<SharedString>(stringId);
+		string1.insertText(0, "hello");
+
+		const waitForSummary = async () => {
+			await new Promise<void>((resolve, reject) => {
+				let summarized = false;
+				container1.on("op", (op) => {
+					if (op.type === "summarize") {
+						summarized = true;
+					} else if (summarized && op.type === "summaryAck") {
+						resolve();
+					} else if (op.type === "summaryNack") {
+						reject(new Error("summaryNack"));
+					}
+				});
+			});
+		};
+		return {
+			loader,
+			provider,
+			container1,
+			map1,
+			cell1,
+			directory1,
+			matrix1,
+			register1,
+			queue1,
+			string1,
+		};
+	}
+
+	it.skip("ensure single attach op sent in map", async function () {
```

Comment:
@anthony-murphy do we want to leave these tests in as well, or just the below tests in the loop?

Reply:
i don't think we need them, as the loop makes them redundent